### DISCUSSION TITLE
TASM Compiler

### DISCUSSION THREAD

**Discussion by abaffa** [hlorenzi/customasm#87] [Ideas]
Hi there!

 I'm looking for a new compiler to replace TASM and I found ur project. My problem is that TASM allows u to create ur own assembly syntax and all new compilers are x386 only... I'm creating a custom homebrew minicomputer using TTL (digital eletronics and chips) and I've projected my own assembly.

There are some dynamic allocations on TASM that ur compiler currently can't do like strings em other commands  (.db, .equ, .dw, etc)... Do u think it's possible to add them?

There is no modern substitute to TASM and ur compiler could be that one!

Best, 
Augusto Baffa
http://sol-1.baffasoft.com.br
http://sol-1.org

**Comment by hlorenzi** [comment]
Hi! I'm glad you're finding the project useful!

What kind of functionality do you need exactly? If I understand correctly, `.db`, `.dw`, and strings can be accomplished by `#d` directives, and you can [read about them here](https://github.com/hlorenzi/customasm/wiki/Outputting-data-blocks-%E2%80%94-%23d). `.equ` can be accomplished with an `=` sign, [as discussed here](https://github.com/hlorenzi/customasm/wiki/Declaring-labels-and-constants).

Do you think these features cover your use cases?

**Reply by abaffa** [reply]
Thanks for ur quick response :D ... I tested on the online compiler and I can check if replacing my code with "#d" and "=" works... but I also noticed that inline strings like " cmp ah, 'n' " didnt work. I'd prefer not to create a label in cases like that.

Does customasm have support for that?

**Reply by hlorenzi** [reply]
Ah, yes! You can just use plain strings with double quotation marks in that case: `cmp ah, "n"` (it will be interpreted as an integer using its UTF-8 encoding.)

**Reply by hlorenzi** [reply]
@abaffa How's it going? 😃 Were you able to make it work as you intended?
